.circleci/config.yml

```diff
@@ -47,9 +47,67 @@ jobs:
           paths:
             - /go/pkg/mod
 
+  benchmarks:
+    parameters:
+      no_output_timeout:
+        type: string
+        description: the job times out after this duration with no output
+        default: 30m
+    environment:
+      GO_VERSION: "1.16"
+      CONSUL_VERSION: "1.8.0"
+      TERRAFORM_VERSION: "0.14.10"
+    docker:
+      - image: docker.mirror.hashicorp.services/circleci/golang:${GO_VERSION}
+    working_directory: /go/src/github.com/{{ORG_NAME}}/{{REPO_NAME}}
+    steps:
+      - checkout
+      - run:
+          name: Install Consul
+          command: |
+            curl -sLo consul.zip https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_linux_amd64.zip
+            unzip consul.zip
+            sudo cp consul /usr/local/bin/
+      - run:
+          name: Install Terraform
+          command: |
+            curl -sLo terraform.zip https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip
+            unzip terraform.zip
+            sudo cp terraform /usr/local/bin/
+      - run:
+          name: Copy terraform to benchmark directory
+          command: /bin/cp /usr/local/bin/terraform ./e2e/
+      - run:
+          name: Version Check
+          command: |
+            go version
+            consul version
+            terraform version
+      - run:
+          name: Fetch Go modules
+          command: go mod download
+      - run:
+          name: Run benchmark suite
+          no_output_timeout: << parameters.no_output_timeout >>
+          command:
+            go test -json ./e2e/benchmarks -timeout 2h -bench=. -tags e2e | /usr/bin/tee /tmp/benchmarks.json
```

Comment:
Can this use `gotestsum` and upload to CircleCI instead to have the tests be consumable by CircleCI? This will allow using test insights and will highlight failures in the UI quicker than scanning a json doc.   

ex:
```
gotestsum --format=short-verbose --junitfile $TEST_RESULTS_DIR/gotestsum-report.xml -- ./e2e/benchmarks -timeout 2h -bench=. -tags e2e
``` 
(untested)

and
```
      - store_test_results:
          path: *TEST_RESULTS_DIR
```

Reply:
I'm not a gotestsum guru, but I didn't think it handled benchmark data.  Have you seen it used to process benchmarks?

Reply:
I just tried this:  `$ gotestsum --format=short-verbose --junitfile moo.xml -- ./e2e/benchmarks/ -timeout 2h -bench=BenchmarkTasksConcurrent_t01_s50 -tags e2e` (limited to this one test because the whole suite takes a while).
I see this in the output: `BenchmarkTasksConcurrent_t01_s50/once_mode-4                   1        31854419398 ns/op`
But the results file doesn't have any benchmark data:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
        <testsuite tests="0" failures="0" time="0.000000" name="github.com/hashicorp/consul-terraform-sync/e2e/benchmarks">
                <properties>
                        <property name="go.version" value="go1.16.2 darwin/amd64"></property>
                </properties>
        </testsuite>
</testsuites>
```

Reply:
Is that related to https://github.com/gotestyourself/gotestsum/issues/62? Does `--format=standard-verbose` like what was suggested in the issue, change anything? If not then it may just not work. I haven't tested it myself but thought it might work so I suggested it 😄

Reply:
Using `standard-verbose` doesn't include the benchmark data either.  So it's probably the test2json issue you linked.